fetch/fetch.go

```diff
@@ -956,7 +958,7 @@ func (f *Fetch) getHash(
 			hint:      h,
 			validator: receiver,
 			promise: &promise{
-				completed: make(chan struct{}, 1),
+				completed: make(chan struct{}),
```

Comment:
removed the buffered channel here because it isn't used anywhere. the channel is only closed.